src/ExEss/Cms/Http/EventSubscriber/LogSubscriber.php

```diff
@@ -0,0 +1,145 @@
+<?php declare(strict_types=1);
+
+namespace ExEss\Cms\Http\EventSubscriber;
+
+use ExEss\Cms\Api\V8_Custom\Service\DataCleaner;
+use ExEss\Cms\FLW_Flows\Request\FlowAction;
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\StreamInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Bridge\PsrHttpMessage\Factory\PsrHttpFactory;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class LogSubscriber implements EventSubscriberInterface
+{
+    private const LOG_NO_CONTENT = 'no_content';
+    private const LOG_BODY = 'body';
+    private const LOG_MODEL = 'model';
+
+    public const MAX_MESSAGE_LENGTH = 10000;
+    public const SUFFIX_TRUNCATED = '... (truncated)';
+
+    private LoggerInterface $logger;
+
+    public function __construct(
+        LoggerInterface $requestLogger
+    ) {
+        $this->logger = $requestLogger;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            // as last during request event
+            KernelEvents::REQUEST => ['logRequest', -1001],
+            // after FlashMessageSubscriber but before StreamedResponseListener
+            KernelEvents::RESPONSE => ['logResponse', -1001],
+        ];
+    }
+
+    private function getPsrRequest(Request $request): ServerRequestInterface
+    {
+        $psr17Factory = new Psr17Factory();
+        $psrHttpFactory = new PsrHttpFactory($psr17Factory, $psr17Factory, $psr17Factory, $psr17Factory);
+
+        return $psrHttpFactory->createRequest($request);
+    }
+
+    private function getPsrResponse(Response $response): ResponseInterface
+    {
+        $psr17Factory = new Psr17Factory();
```

Comment:
we use this on two places now, maybe we can make it a service

Reply:
I thought the same, but ofc the fallback one will be gone, will revisit later on